src/util/TypeTraits.h

```diff
@@ -390,6 +423,12 @@ struct UniqueTypesTupleImpl<T, List...> {
 template <typename... List>
 using UniqueVariant = typename detail::TupleToVariantImpl<
     typename detail::UniqueTypesTupleImpl<List...>::type>::type;
+/// Examples:
+static_assert(std::is_same_v<UniqueVariant<int, int, int, bool>,
+                             std::variant<int, bool>>);
+static_assert(
+    std::is_same_v<UniqueVariant<int, bool>, std::variant<int, bool>>);
+static_assert(std::is_same_v<UniqueVariant<>, std::variant<>>);
```

Comment:
Great that this is the PR where I suggest changes, then we can merge the prerequisites first over night hopefully.